### ISSUE TITLE
Running identical code multiple times has some `UnresolvedValue` results

### ISSUE THREAD

**Issue by James-G-Hill** [shikokuchuo/mirai#174]
Hello, I've been continuing my experimentation with `mirai` and `shiny::ExtendedTask`. I'm encountering a large number of `UnresolvedValue` results and I'm unsure how to interpret this. I've looked through the documentation but it's not entirely clear to me how to understand this. I'll explain what I'm doing.

I have some long running code that exists in an `R6` class. I create a new extended task and pass in the `R6` class; then when I invoke the extended task gets triggered and the code within the `R6` class gets called to return a new object.

I have a `shiny::bindEvent` that requires the `status` of the extended task to change to either `error` or `success` before continuing a `shiny::obseve` that takes the `result` of the extended task and either handles an error or continues with the app functionality.

This now works fine . . . the majority of the time. But a still relatively large proportion of times I receive an `UnresolvedValue` instead of the new object that I would expect from a `success` result. I've used `browser` to check and when this is happening the extended task has definitely reported a successful completion rather than an `error`.

So I'm trying to understand what situations will return an `UnresolvedValue` from a `mirai` that hasn't obviously had an error? Or how does the `mirai` interact with the `success` status from an `ExtendedTask`?

**Comment by shikokuchuo** [comment]
Can you make a simplified version of your code to reproduce what you’re seeing?

When you say you get an ‘unresolvedValue’, does `unresolved()` return `TRUE` on this object? If so then this means that the mirai really has not resolved. Another easy test is if you then reset daemons by a `daemons(0)`, these should now resolve to a connection reset error value.

**Comment by James-G-Hill** [comment]
The number of these has reduced but I will continue testing and try to find case that I can solidly replicate.

**Comment by shikokuchuo** [comment]
Can you check if you still get these with the latest dev version of mirai 1.3.1.9022?

As I mentioned above, if you run `unresolved()` on them, that will let you know if there are actually unresolved. If they are not unresolved then they are values that have been copied across somehow, and just look like unresolved values.

**Comment by shikokuchuo** [comment]
@James-G-Hill I'm assuming you've got to the bottom of this? Let me know if there's something you think I should take a look at.

**Comment by James-G-Hill** [comment]
I think, but not entirely sure and haven't had a chance to test, that this may have been a result of me having an error I'm the glue string passed as the message to the stop commands I was using at the time.

**Comment by James-G-Hill** [comment]
Hello, apologies, I know I've kept you waiting on this. I haven't reduced all possible occurrences of this but a large number of them were the result of my not carefully handling multiple clicks on action buttons and the changing of inputs within my reactive expressions. If I ever come across something repeatable that can't be explained I'll create a new ticket.

**Comment by shikokuchuo** [comment]
Sure, no worries. I'm closing this for you, and hopefully you won't need to come back!
